lib/pbench/server/db/models/tracker.py

```diff
@@ -0,0 +1,565 @@
+        component is mutating the dataset in some way, and a mixed
+        case "friendly" name for the state. "Mutating" states are
+        expected to be transient as the component should complete
+        the mutation, and should usually have "-ing" endings.
+
+        Args:
+            name (string): Friendly name for the state
+            mutating (boolean): True if a component is mutating dataset
+        """
+        self.friendly = friendly
+        self.mutating = mutating
+
+    def __str__(self):
+        "Return the state's friendly name"
+        return self.friendly
+
+
+class Dataset(Database.Base):
+    """ Identify a Pbench dataset (tarball plus metadata)
+
+    Columns:
+        id          Generated unique ID of table row
+        user        User associated with the dataset
+        controller  Name of controller node
+        name        Base name of dataset (tarball)
+        md5         The dataset MD5 hash (Elasticsearch ID)
+        created     Date the dataset was PUT to server
+        state       The current state of the dataset
+        transition  The timestamp of the last state transition
+    """
+
+    __tablename__ = "dataset"
+
+    transitions = {
+        States.UPLOADING: [States.UPLOADED, States.QUARANTINED],
+        States.UPLOADED: [States.UNPACKING, States.QUARANTINED],
+        States.UNPACKING: [States.UNPACKED, States.QUARANTINED],
+        States.UNPACKED: [States.INDEXING, States.QUARANTINED],
+        States.INDEXING: [States.INDEXED, States.QUARANTINED],
+        States.INDEXED: [States.INDEXING, States.EXPIRING, States.QUARANTINED],
+        States.EXPIRING: [States.EXPIRED, States.INDEXED, States.QUARANTINED],
+        # NOTE: EXPIRED and QUARANTINED do not appear as keys in the table
+        # because they're terminal states that cannot be exited.
+    }
+
+    DEFAULT_USERNAME = "public"
+
+    id = Column(Integer, primary_key=True, autoincrement=True)
+    user = Column(String(255), unique=False, nullable=False)
+    controller = Column(String(255), unique=False, nullable=False)
+    name = Column(String(255), unique=False, nullable=False)
+    md5 = Column(String(255), unique=False, nullable=True)
+    created = Column(DateTime, nullable=False)
+    state = Column(Enum(States), unique=False, nullable=False)
+    transition = Column(DateTime, nullable=False)
+    metadatas = relationship("Metadata", backref="dataset")
+
+    # Require that the combination of controller and name is unique
+    # NOTE: Ideally, this would be user+controller+name; but an
+    # important purpose here is to allow non-intake components to
+    # find the owning user of a dataset given the file path, so we
+    # can't allow duplicates. In practice, there'd only be an
+    # actual conflict if two users generated the same metric
+    # configuration on the same controller node in the same second,
+    # so we can probably discount this.
+    __table_args__ = (UniqueConstraint("controller", "name"), {})
```

Comment:
So, when we look it up in the database, there is (we hope...) a record there with `user` and `controller` and (tarball) `name`, right?  The intention is that looking up by `controller`+`name` will produce a unique result, which yields the `user`.  So, the "uniqueness constraint" is on the `controller`+`name`...OK, I got it.

But, given that that is the case, then the comment at line 146 is incorrect:  requiring uniqueness for `user`+`controller`+`name` would allow the possibility for multiple `user` values corresponding to a given `controller`+`name` pair, which would be far from "ideal", even if it is highly unlikely.

I think you should boil this comment down to:

```
    # Require that the combination of controller and name be unique
    # in order to allow non-intake components to find the owning user
    # of a dataset given the file path.
```